Google.Api.Gax.Grpc/Rest/PartialDecodingStreamReader.cs

```diff
@@ -220,28 +203,26 @@ private async Task ReadAndProcessData(CancellationToken cancellationToken)
         }
         // If *anything* fails, this signals an overall failure.
         // Let the calling code effectively rethrow. (We lose the stack trace, which isn't ideal, admittedly.)
-        catch (Exception e)
+        catch (Exception e) when (e is not OperationCanceledException)
         {
             _errorMessage = e.Message;
         }
     }
 
-    private void MaybeDispose()
-    {
-        _textReader?.Dispose();
-        // This would trigger fetching again if we ended up back in ReadAndProcessData,
-        // but we only dispose on terminal states.
-        _textReader = null;
-    }
-
     /// <summary>
-    /// Returns the value from the original task provided in <paramref name="task"/>, or <paramref name="defaultValue"/>
+    /// Returns the value from the original task provided in <paramref name="task"/>, or throws
     /// if <paramref name="cancellationToken"/> was cancelled before the task completed.
     /// </summary>
-    private static async Task<T> AwaitWithCancellation<T>(Task<T> task, CancellationToken cancellationToken, T defaultValue)
+    private static async Task<T> AwaitWithCancellation<T>(Task<T> task, CancellationToken cancellationToken)
     {
         var cancellationTask = Task.Delay(-1, cancellationToken);
         var resultTask = await Task.WhenAny(task, cancellationTask).ConfigureAwait(false);
-        return resultTask == cancellationTask ? defaultValue : task.Result;
+        if (resultTask == cancellationTask)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            // This is just a backstop. We really shouldn't get here.
+            throw new InvalidOperationException("Cancelled token did not throw");
+        }
+        return task.Result;
```

Comment:
```suggestion
        cancellationToken.ThrowIfCancellationRequested();
        return task.Result;
```

Reply:
That code is certainly simpler, yes. I'm probably being paranoid about the cancelled task being done without the cancellation token being cancelled. Will do this, but with an additional comment.